src/matrixCells/makeMatrixArgs.js

```diff
@@ -64,27 +98,28 @@ export default function make_matrix_args(regl, store) {
                         );
 
       // pass attribute (in vert) to varying in frag
-      opacity_vary = opacity_att;
-
+      intensity_vary = intensity_arr;
+      mask_vary = mask_arr;
     }`;
   const frag_string = `
     // precision highp float;
     precision lowp float;
-
     uniform vec3 pos_rgb;
     uniform vec3 neg_rgb;
-
     // use the varying being passed from the vertex shader
-    varying float opacity_vary;
-
+    varying float intensity_vary;
+    varying float mask_vary;
     void main() {
-
-      if (opacity_vary > 0.0){
-        // gl_FragColor = vec4(1, 0, 0, abs(opacity_vary));
-        gl_FragColor = vec4(pos_rgb, abs(opacity_vary));
+      if (intensity_vary > 0.0){
+        gl_FragColor = vec4(pos_rgb, abs(intensity_vary));
       } else {
-        // gl_FragColor = vec4(0, 0, 1, abs(opacity_vary));
-        gl_FragColor = vec4(neg_rgb, abs(opacity_vary));
+        gl_FragColor = vec4(neg_rgb, abs(intensity_vary));
+      }
+      if (mask_vary == 0.3) {
+        gl_FragColor = vec4(1.0, 1.0, 0.0, min(1.0, intensity_vary + 0.3));
+      }
+      if (mask_vary > 0.4) {
+        gl_FragColor = vec4(1.0, 0.0, 0.0, min(1.0, intensity_vary + 0.5));
```

Comment:
It looks like this is replacing red/blue if the mask is either equal to `0.3` or greater than `0.4`, is that the intention? I sort of figured that only the opacity/intensity would change, is that not correct?

Reply:
values equal to `0.3` are the highlight zones.
values greater than `0.4` are the intersected zones.

Reply:
How does this work in regards to `neg_rgb` and `pos_rgb`? It looks like you're not using them here? Does the correct color still show?

Reply:
We simply overwrite the color if it's selected. But we take the intensity value into accounts but amplify them. So that users could see the highlight with original patterns